src/sentry/tasks/relay.py

```diff
@@ -155,13 +154,81 @@ def schedule_update_config_cache(
         update_reason=update_reason,
     )
 
+    # Checking if the project is debounced and debouncing it are two separate
+    # actions that aren't atomic. If the process marks a project as debounced
+    # and dies before scheduling it, the cache will be stale for the whole TTL.
+    # To avoid that, make sure we first schedule the task, and only then mark
+    # the project as debounced.
+    projectconfig_debounce_cache.debounce(public_key, project_id, organization_id)
+
+
+@instrumented_task(
+    name="sentry.tasks.relay.build_project_config",
+    queue="relay_config",
+    acks_late=True,
+    soft_time_limit=TASK_SOFT_LIMIT,
+    time_limit=TASK_HARD_LIMIT,
+)
+def build_project_config(public_key=None, trigger=None, **kwargs):
+    """Build a project config and put it in the Redis cache.
+
+    This task is used to compute missing project configs, it is aggressively
+    deduplicated to avoid running duplicate computations and thus should only
+    be invoked using :func:`schedule_build_project_config`. Because of this
+    deduplication it is not suitable for re-computing a project config when
+    an option changed, use :func:`schedule_invalidate_project_config` for this.
+
+    Do not invoke this task directly, instead use :func:`schedule_build_project_config`.
+    """
+
+    try:
+        validate_args(public_key=public_key)
+        sentry_sdk.set_tag("update_reason", trigger)
```

Comment:
I think we need to set a few more tags like the invalidation task does as well: org_id, proj_id, pub_key.  before this was done inside `validate_args` but that was problematic in CI so was removed.

Reply:
Why was it problematic in CI? I'm curious.